chainio/clients/fireblocks/client.go

```diff
@@ -0,0 +1,162 @@
+package fireblocks
+
+import (
+	"bytes"
+	"context"
+	"crypto/rsa"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/golang-jwt/jwt"
+	"github.com/google/uuid"
+)
+
+type FireblocksTxID string
+
+type FireblocksClient interface {
+	ContractCall(ctx context.Context, body *ContractCallRequest) (*ContractCallResponse, error)
+	ListContracts(ctx context.Context) ([]WhitelistedContract, error)
+}
+
+type fireblocksClient struct {
+	apiKey     string
+	privateKey *rsa.PrivateKey
+	baseURL    string
+	timeout    time.Duration
+	client     *http.Client
+	logger     logging.Logger
+}
+
+type Asset struct {
+	ID      string         `json:"id"`
+	Status  string         `json:"status"`
+	Address common.Address `json:"address"`
+	Tag     string         `json:"tag"`
+}
+
+type ErrorResponse struct {
+	Message string `json:"message"`
+	Code    int    `json:"code"`
+}
+
+type WhitelistedContract struct {
+	ID     string  `json:"id"`
+	Name   string  `json:"name"`
+	Assets []Asset `json:"assets"`
+}
+
+func NewFireblocksClient(apiKey string, secretKey []byte, baseURL string, timeout time.Duration, logger logging.Logger) (FireblocksClient, error) {
+	c := http.Client{Timeout: timeout}
+	privateKey, err := jwt.ParseRSAPrivateKeyFromPEM(secretKey)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing RSA private key: %w", err)
```

Comment:
Did a quick search. Seems like `errors.Join()` is the new way since go 1.20 to combine errors (see for eg https://lukas.zapletalovi.com/posts/2022/wrapping-multiple-errors/). Wrapping would wrap errors (one contains the other), whereas join creates an error with a list of error strings. Still not 100% sure what the difference is, but it seems like join is easier to deal with and implement `is` to check if a combined error contains some error.